test/feature-benchmark/mzcompose.py

```diff
@@ -132,29 +136,19 @@ def run_one_scenario(
                 param_name, param_value = param.split("=")
                 additional_system_parameter_defaults[param_name] = param_value
 
-        mz = Materialized(
-            image=f"materialize/materialized:{tag}" if tag else None,
-            default_size=size,
-            # Avoid clashes with the Kafka sink progress topic across restarts
-            environment_id=f"local-az1-{uuid.uuid4()}-0",
-            soft_assertions=False,
-            additional_system_parameter_defaults=additional_system_parameter_defaults,
-            external_cockroach=True,
-            external_minio=True,
-            sanity_restart=False,
-        )
-
-        with c.override(mz):
-            print(f"The version of the '{instance.upper()}' Mz instance is:")
-            c.run(
-                "materialized",
-                "-c",
-                "environmentd --version | grep environmentd",
-                entrypoint="bash",
-                rm=True,
-            )
+        mz = create_mz_image(tag, size, additional_system_parameter_defaults)
 
-            c.up("cockroach", "materialized")
+        try:
+            start_overridden_mz_and_cockroach(c, mz, instance)
+        except Exception as e:
```

Comment:
This I think is a risky code, as `start_overriden_mz_and_cockroach` may fail for any number of reasons.

Consider calling `docker pull` on the image instead, like this `c.invoke("pull", service, max_tries=2)` and revert to `latest` if this returns a `UIError`

Reply:
Well, I have tried to narrow it down in the `except` clause by limiting the logic to `running docker compose failed (exit status 15)` (but wasn't able to get any more precise error messages).
I am happy to apply your suggestion instead.

✅ Adjusted.

Reply:
@philip-stoev: `c.invoke("pull", ...)` will do a `docker compose pull`, which does not find the image even after a lot of retries. `docker pull` would find it.

Reply:
I'm not sure I understand this problem.